lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -826,6 +826,22 @@ export class NormalComponent<
           const rect = db.schematic_rect.get((child as any).schematic_rect_id)
           if (rect) schematicElements.push(rect)
         }
+        if (
+          child.isSchematicPrimitive &&
+          child.componentName === "SchematicCircle"
+        ) {
+          const circle = db.schematic_circle.get(
+            (child as any).schematic_circle_id,
+          )
+          if (circle) schematicElements.push(circle)
+        }
+        if (
+          child.isSchematicPrimitive &&
+          child.componentName === "SchematicArc"
+        ) {
+          const arc = db.schematic_arc.get((child as any).schematic_arc_id)
+          if (arc) schematicElements.push(arc)
+        }
```

Comment:
this logic is definitely super jank, NormalComponent isn't suppost to have class specific behavior